### PR TITLE
US-398395: OpenSearch image

### PR DESCRIPTION
**Pull request by kusumaghoshdastidar**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=398395

Opensearch v 1.3.2 image on opensuse java 11 base container

For starting the service refer changes here:
https://github.com/opendistro-for-elasticsearch/opendistro-build/blob/main/elasticsearch/docker/build/elasticsearch/bin/docker-entrypoint.sh#L7
vs
https://github.com/opensearch-project/opensearch-build/blob/main/docker/release/config/opensearch/opensearch-docker-entrypoint.sh#L76

### CODE REVIEW

File: src/main/docker/scripts/start.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+export JAVA_HOME=$JRE_HOME
+chown -R opensearch:opensearch /usr/share/opensearch
+exec chroot --userspec=1000 / /usr/share/opensearch/opensearch-docker-entrypoint.sh "opensearch"
```
Comment (dermot-hardy) [inline src/main/docker/scripts/start.sh:20]: Why do we have to use `chroot` to run this?

File: .gitignore [hunk 2]
```diff
@@ -1,23 +1,34 @@
-# Compiled class file
```
[the remark this answers is not in the document]
Comment (kusumaghoshdastidar) [reply .gitignore:1]: log4j-core is being included from com.github.cafapi.logging:caf-logging-log4j2:jar

[the remark this answers is not in the document]
Comment (dermot-hardy) [reply .gitignore:1]: If that's the only reason then no-one's stopping us from using `com.github.cafapi.logging:caf-logging-logback` instead.

[the remark this answers is not in the document]
Comment (kusumaghoshdastidar) [reply .gitignore:1]: OpenSearch is using log4j : https://opensearch.org/docs/latest/opensearch/logs/#application-logs
Tried using logback instead but OpenSearch plugin installation fails:
https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/logging/LogConfigurator.java#L121
`
[INFO] DOCKER> [91mException in thread "main" 
[INFO] DOCKER> [91mjava.lang.NoClassDefFoundError: org/apache/logging/log4j/core/config/properties/PropertiesConfigurationFactory
[INFO] DOCKER> [91m

[INFO] DOCKER> [91m	at org.opensearch.cli.CommandLoggingConfigurator.configureLoggingWithoutConfig(CommandLoggingConfigurator.java:53)
	at org.opensearch.cli.Command.main(Command.java:98)

[INFO] DOCKER> [91m	at org.opensearch.plugins.PluginCli.main(PluginCli.java:60)

[INFO] DOCKER> [91mCaused by: java.lang.ClassNotFoundException: org.apache.logging.log4j.core.config.properties.PropertiesConfigurationFactory
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
[INFO] DOCKER> [91m
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
`

File: src/main/docker/scripts/start.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+export JAVA_HOME=$JRE_HOME
+chown -R opensearch:opensearch /usr/share/opensearch
```
[the remark this answers is not in the document]
Comment (kusumaghoshdastidar) [reply src/main/docker/scripts/start.sh:19]: Some configuration files are overwritten by the classification worker and their ownership needed to be changed, else building the classification worker on Jenkins throws permission errors. Moving this to the dockerfile didn't help.

File: src/main/docker/scripts/start.sh (hunk 1 above)
Comment (kusumaghoshdastidar) [reply src/main/docker/scripts/start.sh:20]: Followed this example https://github.com/opendistro-for-elasticsearch/opendistro-build/blob/main/elasticsearch/docker/build/elasticsearch/bin/docker-entrypoint.sh#L7
I tried setting RUNAS_USER to start OpenSearch as a non-root user, but see permission errors when building classification worker.

File: src/test/java/cafapi/opensuse/opensearch1/ContainerIT.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+            final CreateIndexResponse createIndexResponse = client.indices().create(createIndexRequest);
+
+            assertTrue("Index response was not acknowledged", createIndexResponse.acknowledged());
+            assertTrue("All shards were not copied", createIndexResponse.shardsAcknowledged());
+        }
+    }
+
+    private OpenSearchTransport getOpenSearchTransport(final RestClient restClient) {
+        LOGGER.info("Creating OpenSearchTransport...");
```
[the remark this answers is not in the document]
Comment (kusumaghoshdastidar) [reply src/test/java/cafapi/opensuse/opensearch1/ContainerIT.java]: Got past this warning in the healthcheck, but can't figure out what is causing it during index creation. Tried a few variations of making the index request but the warning is still logged.

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**